cluster_config_test.go

```diff
@@ -111,20 +112,51 @@ func TestLoadJSON(t *testing.T) {
 
 	j = &configJSON{}
 	json.Unmarshal(ccfgTestJSON, j)
-	j.StateSyncInterval = ""
+	j.ReplicationFactor = 0
+	j.ReplicationFactorMin = 0
+	j.ReplicationFactorMax = 0
+	tst, _ = json.Marshal(j)
+	cfg.LoadJSON(tst)
+	if cfg.ReplicationFactorMin != -1 || cfg.ReplicationFactorMax != -1 {
+		t.Error("expected default replication factor")
+	}
+
+	j = &configJSON{}
+	json.Unmarshal(ccfgTestJSON, j)
+	j.ReplicationFactor = 3
+	tst, _ = json.Marshal(j)
+	cfg.LoadJSON(tst)
+	if cfg.ReplicationFactorMin != 3 || cfg.ReplicationFactorMax != 3 {
+		t.Error("expected replicationFactor Min/Max override")
+	}
+
+	j = &configJSON{}
+	json.Unmarshal(ccfgTestJSON, j)
+	j.ReplicationFactorMin = -1
 	tst, _ = json.Marshal(j)
 	err = cfg.LoadJSON(tst)
 	if err == nil {
-		t.Error("expected error state_sync_interval")
+		t.Error("expected error when only one replication factor is -1")
 	}
 
 	j = &configJSON{}
 	json.Unmarshal(ccfgTestJSON, j)
-	j.ReplicationFactor = 0
+	j.ReplicationFactorMin = 5
+	j.ReplicationFactorMax = 4
 	tst, _ = json.Marshal(j)
-	cfg.LoadJSON(tst)
-	if cfg.ReplicationFactor != -1 {
-		t.Error("expected default replication factor")
+	err = cfg.LoadJSON(tst)
+	if err == nil {
+		t.Error("expected error when only rplMin > rplMax")
+	}
+
+	j = &configJSON{}
+	json.Unmarshal(ccfgTestJSON, j)
+	j.ReplicationFactorMin = 0
+	j.ReplicationFactorMax = 0
+	tst, _ = json.Marshal(j)
+	err = cfg.LoadJSON(tst)
+	if err == nil {
+		t.Error("expected error when only rplMin/Max is not set")
```

Comment:
small issue: why use the word "only", it makes it sound like it is ok for rplMin/Max to be unset when something else is set.  I didn't see this from the code above, though perhaps I missed something.